### PR TITLE
Issue #44

### LINKED ISSUE #44: Play all on the page - Youtube embedded bug
Replication:
1. Get on a page with multiple Youtube embedded clips
2. Play one.
3. Pause it.
4. Press Stoplay from toolbar.
5. All the clips start to play.

### PR DESCRIPTION
**Pull request by NffS**
Fixes #44

### CODE REVIEW

**Comment by beshur** [comment]
Looks fine, testing locally for a couple of days.

File: background.js [hunk 1]
```diff
@@ -158,26 +158,33 @@ chrome.storage.onChanged.addListener(function(changes, namespace) {
 
 chrome.browserAction.onClicked.addListener(function(e) {
 	var lastPlayingTabId = parseInt(DataStorage.get('lastPlayingTabId')),
+        lastPlayingFrameId = parseInt(DataStorage.get('lastPlayingFrameId')),
```
Comment (endway) [inline background.js]: `lastPlayingFrameId = parseInt(DataStorage.get('lastPlayingFrameId')) | 0`

File: background.js [hunk 2]
```diff
@@ -158,26 +158,33 @@ chrome.storage.onChanged.addListener(function(changes, namespace) {
...
 		lastPausedTabId = parseInt(DataStorage.get('lastPausedTabId')),
+        lastPausedFrameId = parseInt(DataStorage.get('lastPausedFrameId')),
```
Comment (endway) [inline background.js]: `lastPausedFrameId = parseInt(DataStorage.get('lastPausedFrameId')) | 0`
this way we don't need check it then

File: background.js [hunk 3]
```diff
@@ -191,23 +198,26 @@ chrome.runtime.onMessage.addListener(function(request, sender, sendResponse) {
 				break;
 
 			case 'started':
-				if(lastPlayingTabId && sender.tab.id != lastPlayingTabId) {
-					chrome.tabs.sendMessage(lastPlayingTabId, {action: 'pause'});
+				var isFrameIdChanged = (lastPlayingTabId && lastPlayingFrameId && sender.frameId != lastPlayingFrameId);
```
Comment (endway) [inline background.js]: we can have here lastPlayingFrameId equal to 0

File: background.js [hunk 4]
```diff
@@ -158,24 +158,20 @@ chrome.storage.onChanged.addListener(function(changes, namespace) {
 
 chrome.browserAction.onClicked.addListener(function(e) {
 	var lastPlayingTabId = parseInt(DataStorage.get('lastPlayingTabId')),
-        lastPlayingFrameId = parseInt(DataStorage.get('lastPlayingFrameId')),
+        lastPlayingFrameId = parseInt(DataStorage.get('lastPlayingFrameId')) | 0,
```
[the remark this answers is not in the document]
Comment (endway) [reply background.js]: My fault, I mentioned bitwise operator in comments. But in this case, it should probably work as well.

[the remark this answers is not in the document]
Comment (beshur) [reply background.js]: Guys, let's keep it simple.
OR is not the same as bitwise OR.

**Comment by beshur** [comment]
However #71 still not fixed by commit 0e39972
Let's continue fixing in a separate branch

**Pull request merged**